sn_registers/src/register.rs

```diff
@@ -166,16 +145,25 @@ impl RegisterReplica {
         self.crdt.merge(other.crdt);
     }
 
+    /// Check if a register op is valid for our current register
+    pub fn check_register_op(&self, op: &RegisterOp) -> Result<()> {
+        self.check_user_rights(Action::Write, op.source)?;
+
+        match op.source {
+            User::Anyone => Ok(()),
+            User::Key(pk) => op.verify_signature(&pk),
+        }
+    }
+
     /// Helper to check permissions for given `action`
     /// for the given requester's public key.
     ///
     /// Returns:
     /// `Ok(())` if the permissions are valid,
     /// `Err::AccessDenied` if the action is not allowed.
-    pub fn check_permissions(&self, action: Action, requester: Option<User>) -> Result<()> {
-        let requester = requester.unwrap_or(self.authority);
+    pub fn check_user_rights(&self, action: Action, requester: User) -> Result<()> {
         // First checks if the requester is the owner.
-        if action == Action::Read || requester == self.policy.owner {
+        if action == Action::Read || requester == self.owner {
             Ok(())
         } else {
```

Comment:
I'm wondering if we shouldn't give all rights to the owner?

Reply:
As they can effectively update to give themselves this, it does seem like they may as well just have perms there already, yeh 👍